### DISCUSSION TITLE
Output of bash command substitution?

### DISCUSSION THREAD

**Discussion by andreas-hakansson-ingka** [community/community#26547] [Actions]
<p>Hi,</p>
<p>In our project, we have a Makefile, with a step to build our docker container. The real version is a bit more complex than the example below, but it captures the gist of what we’re doing.</p>
<pre><code class="lang-auto">docker-build:
	@echo "🚀 Building docker image"
	@docker build \
		-t ${DOCKER_IMAGE} .
	@echo ${DOCKER_IMAGE}
</code></pre>
<p><code>${DOCKER_IMAGE}</code> is calculated based on commit SHA etc. Part of this is that we want to capture the calculated container name + tag, so we echo <code>${DOCKER_IMAGE}</code> and capture it in our github action like this</p>
<pre><code class="lang-yaml">- name: Docker (build)
  id: container
  run: |
    label=$(make docker-build | tail -1)
    echo "::set-output name=label::$label"

- name: Display container SHA
  run: echo ${{ steps.container.outputs.label }}
</code></pre>
<p>Which works great. The problem is that since we need to capture output of the make recipe, we need to call it using a bash command substitution <code>$(make docker-build | tail -1)</code> however, when we do this we do not get any output of the <code>docker build</code> command, which would make troubleshooting difficult if we ever run into problems.</p>
<p>The exact same setup works fine locally on our machines, but the stdout from the inner <code>docker build</code> is suppressed by GitHub Actions.</p>
<p>Does anyone have suggestions on how we could continue capturing the output, while also getting the output of the <code>docker build</code> command to be displayed?</p>
<p>Thanks!</p>

**Comment by airtower-luna** [comment]
<p>How about writing <code>DOCKER_IMAGE</code> to a file from the Makefile, instead of capturing output? Later steps could then read it from that file, and the only “cost” to local builds is one tiny additional output file. :slightly_smiling_face:</p>

**Comment by andreas-hakansson-ingka** [comment]
<p>To be honest I’d like to avoid creating a bunch of artifacts from my Makefile since we run them locally as well. Like I said, this works flawless locally since it’s not Make that’s hiding the output, it seems to be GitHub Actions</p>
<p>Running <code>label=$(make docker-build | tail -1)</code> locally both gets me the output as well as the build log of my Docker image</p>

**Comment by airtower-luna** [comment]
<p>Then <em>copy</em> the output to a file in the Actions workflow before processing it, something like this:</p>
<pre><code class="lang-bash">make docker-build | tee output.log
label=$(tail -1 output.log)
</code></pre>

**Comment by andreas-hakansson-ingka** [comment, marked as answer]
<p>I decided to go another way. Since the generated container image name (and tag) is idempotent for the same commit, I decided to add a <code>docker-info</code> recipe in my Makefile. I simply call that to get the name</p>
